src/client/Microsoft.Identity.Client/Platforms/Android/Broker/AndroidBrokerHelper.cs

```diff
@@ -318,7 +318,27 @@ public async Task InitiateBrokerHandshakeAsync(Activity callerActivity)
                             return;
                         }
 
-                        throw new MsalClientException("Could not negotiate protocol version with broker.");
+                        dynamic errorResult = JObject.Parse(bundleResult?.GetString(BrokerConstants.BrokerResultV2));
+                        string error = null;
+                        string errorDescription = null;
+
+                        if (errorResult != null)
+                        {
+                            error = errorResult[BrokerResponseConst.BrokerErrorCode]?.ToString();
+                            errorDescription = errorResult[BrokerResponseConst.BrokerErrorMessage]?.ToString();
+
+                            _logger.Error($"error: {error} errorDescription {errorDescription}");
```

Comment:
>e [](start = 44, length = 1)

nit: capitalize E in error and errorDescription

Reply:
does the error message from broker contain any PII? 

---
In reply to: [490529201](https://github.com/AzureAD/microsoft-authentication-library-for-dotnet/pull/2047#discussion_r490529201) [](ancestors = 490529201)